text/0129-video-replay-envelope.md

```diff
@@ -0,0 +1,62 @@
+- Start Date: 2024-02-06
+- RFC Type: feature
+- RFC PR: [#129](https://github.com/getsentry/rfcs/pull/129)
+- RFC Status: draft
+
+# Summary
+
+In order to capture video replays and present them to the user, we need to define/decide:
+
+- how to transport the video data to the server
+- how to integrate the video data into the RRWeb JSON format
+- how to combine multiple replay chunks to a single replay session
+
+All of these influence one another and need to be considered together in a single RFC.
+
+Note: SDK-side implementation that is currently being worked on relies on taking screenshots and encoding them to a video.
+This is based on an evaluation where a video has much smaller size than a sequence of images (**TODO fix these: factor of X for 720p video**).
+
+# Motivation
+
+We need this to to capture replays on platforms where it's not possible/feasible to produce an HTML DOM (i.e. the native format supported by RRWeb). For example: mobile apps.
+
+<!-- # Supporting Data -->
+<!-- Metrics to help support your decision (if applicable). -->
+
+# Options Considered
+
+## Using a video, with EnvelopeItem:Attachment
+
+- From the SDK, we should be able to use the existing envelope item type, [`ReplayRecording`](https://github.com/getsentry/relay/blob/5fd3969e88d3eea1f2849e55b61678cac6b14e44/relay-server/src/envelope.rs#L115C5-L115C20) to transport the video data.
+- Additionally, it would be accompanied with replay envelope item type, [`ReplayEvent`](https://github.com/getsentry/relay/blob/5fd3969e88d3eea1f2849e55b61678cac6b14e44/relay-server/src/envelope.rs#L113) in RRWeb JSON format.
```

Comment:
Do not send RRWeb formatted JSON on the `ReplayEvent` envelope type.  Please refer to the protocol defined for the `ReplayEvent` here: https://github.com/getsentry/relay/blob/8411b1dfaaee645658cd5e9d5428058f2b19384f/relay-event-schema/src/protocol/replay.rs#L26

The information on this event is required to enable search and sort.

Reply:
Okay, how **do** we send the actual replay JSON data then? This thing ought to have some docs...